### PR TITLE
fix(indev): fix elastic scrolling with snapping

### PR DESCRIPTION
**Pull request by kisvegabor**
Earlier it was depending on the pressed object if elastic scroll is applied with snapping. Now it explicitly checks if there are more snap points and if not it switches to elastic scroll

### Description of the feature or fix

A clear and concise description of what the bug or new feature is.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` ([astyle](http://astyle.sourceforge.net/install.html) version [v3.4.12](https://github.com/szepeviktor/astyle/releases/tag/v3.4.12) needs to be installed) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: src/indev/lv_indev_scroll.c [hunk 1]
```diff
@@ -500,17 +548,18 @@ static int32_t find_snap_point_x(const lv_obj_t * obj, int32_t min, int32_t max,
         }
     }
 
-    return dist == LV_COORD_MAX ? 0 : -dist;
+    return dist == LV_COORD_MAX ? LV_COORD_MAX : -dist;
```
Comment (liamHowatt) [inline src/indev/lv_indev_scroll.c:550]: This function has another `return` in it. Should the other `return` be updated to return `LV_COORD_MAX` too? FYI, that other `return` -- taken when `align == LV_SCROLL_SNAP_NONE` -- will only be reached when this function is called by the public function `lv_indev_scroll_get_snap_dist`.

File: src/indev/lv_indev_scroll.c [hunk 2]
```diff
@@ -555,7 +604,7 @@ static int32_t find_snap_point_y(const lv_obj_t * obj, int32_t min, int32_t max,
...
```
Comment (liamHowatt) [inline src/indev/lv_indev_scroll.c:606]: This function too.

**Review by liamHowatt — DISMISSED** [review]
Seems like solid logic to me!

File: src/indev/lv_indev_scroll.c (hunk 1 above)
Comment (kisvegabor) [reply src/indev/lv_indev_scroll.c:550]: Added.

File: src/indev/lv_indev_scroll.c (hunk 2 above)
Comment (kisvegabor) [reply src/indev/lv_indev_scroll.c:606]: Added.

**Review by liamHowatt — APPROVED** [review]

**Comment by kisvegabor** [comment]
@XuNeo @FASTSHIFT could you review this PR?

**Review by XuNeo — APPROVED** [review]

**Comment by kisvegabor** [comment]
Thank you!

**Pull request merged**